### ISSUE TITLE
[Bug]: `stylesToArray` throws error with custom style

### ISSUE THREAD

**Issue by schnip** [fabricjs/fabric.js#9137]
### CheckList

- [X] I agree to follow this project's Code of Conduct
- [X] I have read and followed the [Contributing Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md)
- [X] I have read and followed the [Issue Tracker Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md#%EF%B8%8F-issue-tracker)
- [X] I have searched and referenced existing issues and discussions
- [X] I am filing a **BUG** report.
- [X] I have managed to reproduce the bug **after** upgrading to the **latest** version
- [X] I have created an accurate and minimal reproduction

### Version

5.3.0

### In What environments are you experiencing the problem?

Firefox, Chrome

### Node Version (if applicable)

None

### Link To Reproduction

https://codesandbox.io/s/bold-brown-pfsvp3?file=/src/index.js:181-192

### Steps To Reproduce

1. Create a new textbox
2. Add a custom style to the text using `setSelectionStyles`
3. Add this textbox to a canvas
4. Call something on the canvas that uses `stylesToArray`, such as `toJSON`

I think the problem is that in the `stylesToArray` function, when `Object.keys(thisStyle).length > 0`, there is an implicit assumption that the first time this happens, it is a style that is recognized in the `hasStyleChanged` function. If the style is one that is not picked up in `hasStyleChanged`, the `stylesArray` will be empty, and the error will happen.

This issue similar to, but unfortunately not a duplicate of https://github.com/fabricjs/fabric.js/issues/8354

### Expected Behavior

No error, the application continues without issue. This is the case with older versions of fabric, such as in the following code sandbox. This particular sandbox is with v5.2.1, but it also works in v4.2.3 (the version I was upgrading from when I discovered this issue).

https://codesandbox.io/s/lingering-silence-9d6485?file=/src/index.js

### Actual Behavior

The attached error is thrown when `stylesToArray` is called

https://codesandbox.io/s/bold-brown-pfsvp3?file=/src/index.js

### Error Message & Stack Trace

```bash
TypeError: stylesArray[(stylesArray.length - 1)] is undefined
    stylesToArray fabric.js:1906
    toObject fabric.js:27409
    addMethods/klass.prototype[property]</< fabric.js:3098
    callSuper fabric.js:3144
    toObject fabric.js:31053
    addMethods/klass.prototype[property]</< fabric.js:3098
    toDatalessObject fabric.js:15339
    _toObject fabric.js:10081
    callSuper fabric.js:3144
    _toObject fabric.js:12889
    addMethods/klass.prototype[property]</< fabric.js:3098
    _toObjects fabric.js:10066
    _toObjects fabric.js:10065
    _toObjectMethod fabric.js:10047
    toDatalessObject fabric.js:10037
    toDatalessJSON fabric.js:10019
    $csb$eval index.js:10
    H eval.js:31
    evaluate transpiled-module.js:715
    evaluateTranspiledModule manager.js:340
    c transpiled-module.js:697
    loadResources index.html:3
    $csb$eval index.html:9
    H eval.js:31
    evaluate transpiled-module.js:715
    evaluateTranspiledModule manager.js:340
    evaluateModule manager.js:311
    e compile.ts:834
    c runtime.js:45
    _invoke runtime.js:274
    r runtime.js:97
```

**Comment by ShaMan123** [comment]
I am able to reproduce
However, it is unclear why you need it or for what reason.
Anyways you should pass an object and not a string, but that isn't repsected also.

**Comment by schnip** [comment]
You are correct about an object being passed instead of a string, I just made a mistake there with the codesandbox

As for use case, the application I'm working on has some custom text annotations in addition to the normal ones. It made sense to store them in the same place, since that seemed to work really well.
